### ISSUE TITLE
Doesn't work without w3m installed

### ISSUE THREAD

**Issue by ieure** [msnyder-info/gopher.el#22]
Because of [this require](https://github.com/msnyder-info/gopher.el/blob/e4ee3284e95e7cdbfe577ab80aac2d4ce872f117/gopher.el#L40), the entire gopher feature fails to load unless you also have w3m-el installed.

I think it would make sense to move the require into `gopher-open-w3m-url`, or break this out into a separate `w3m-gopher.el`.

**Comment by gum3n** [comment]
I just encounter the same issue.

> unless you also have w3m-el installed.

This is not enough. After installing `w3m-el` I have different error `(error "Install w3m command in ‘exec-path’ or set ‘w3m-command’ variable correctly")`. It's because I don't have installed `w3m` on this computer.

**Comment by kensanata** [comment]
I'm a bit confused by the use case for `gopher-open-w3m-url`. Who calls this? Not our code, apparently! So perhaps simply making w3m optional would be enough? Or removing all of it? Are we just trying to silence compiler warnings? Questions, questions.

```diff
modified   gopher.el
@@ -37,7 +37,7 @@
 
 (require 'cl-lib)
 (require 'shr)
-(require 'w3m)
+(require 'w3m nil t); optional
 
 ;;; Code:
 
@@ -405,6 +405,8 @@ Unreliable, e.g. foo.com:70/1/bar becomes foo.com:70/1/, which is useless."
 (defun gopher-open-w3m-url ()
   "Open this URL in Gopher."
   (interactive)
+  (unless (functionp 'w3m-anchor)
+    (error "You need to install the w3m package for this"))
   (gopher (w3m-anchor)))
 
 (defmacro gopher-navigate (content-type &optional reverse)
```

**Comment by gum3n** [comment]
As far as I understand it's function for You to use inside w3m browser. You call it on gopher links inside w3m. So you can easily open them in gopher buffer.

I like idea of making it optional.

**Comment by kensanata** [comment]
What do you think of 98245e0f34190ab3ba430f21ec85935597196949?

**Comment by kensanata** [comment]
Merged 6f4accac226698b22e8388e41ad5723b12553dde. Let me know if we can close this. 😃

**Comment by ieure** [comment]
Looks good, just tried it out and it's working for me.  Thanks for the quick turnaround!
